src/workerd/io/worker-entrypoint.c++

```diff
@@ -213,6 +213,63 @@ void WorkerEntrypoint::init(
       .attach(kj::mv(actor));
 }
 
+/**
+ * Setup Python with a more generous limit enforcer, currently enterStartupJs.
+ *
+ * The Pythons setup is async so we hold the limitScope until the setup work is done. Whatever else
+ * happens between now and when we finish initPython will be done with the higher limit. User code
+ * cannot be entered with the higher limit because the first thing the main entrypoints do is wait
+ * on this initialization work being completed.
+ *
+ * Ideally we should be able to wrap a JS task so that only it is executed with a different limiter,
+ * which is switched on when entering the task and back off when exiting. But I don't know how to do
+ * that so maybe this works okay for now.
+ */
+void
+initPythonIfNeeded(kj::Maybe<kj::StringPtr>& entrypointName, kj::Own<workerd::IoContext_IncomingRequest>& incomingRequest) {
+  auto& context = incomingRequest->getContext();
+  auto& script = context.getWorker().getScript();
+  if (!script.isPython) {
+    return;
```

Comment:
Might be good to throw here

Reply:
We call this function unconditionally hence the `IfNeeded` in `initPythonIfNeeded`. If it's **not** needed, we return without doing anything. If we had to move these 5 lines of code into each of the 5 call sites it would be an extra (5-1)*5 = 20 lines of boilerplate. In the interest of minimizing code duplication I figured it's better to do the check in here.